monkey/monkey_island/cc/repository/mongo_credentials_repository.py

```diff
@@ -0,0 +1,98 @@
+from typing import Sequence
+
+from flask_pymongo import PyMongo
+
+from common.credentials import Credentials
+from monkey_island.cc.repository import RemovalError, RetrievalError, StorageError
+from monkey_island.cc.repository.i_credentials_repository import ICredentialsRepository
+
+
+class MongoCredentialsRepository(ICredentialsRepository):
+    """
+    Store credentials in a mongo database that can be used to propagate around the network.
+    """
+
+    def __init__(self, mongo_db: PyMongo):
+        self._mongo = mongo_db
+
+    def get_configured_credentials(self) -> Sequence[Credentials]:
+        try:
+
+            return MongoCredentialsRepository._get_credentials_from_collection(
+                self._mongo.db.configured_credentials
+            )
+        except Exception as err:
+            raise RetrievalError(err)
+
+    def get_stolen_credentials(self) -> Sequence[Credentials]:
+        try:
+            return MongoCredentialsRepository._get_credentials_from_collection(
+                self._mongo.db.stolen_credentials
+            )
+        except Exception as err:
+            raise RetrievalError(err)
```

Comment:
Move the error handling down into `_get_credentials_from_collection()` to reduce duplication. Do this for the rest of the methods, too.